lib/matroid/connection.rb

```diff
@@ -0,0 +1,125 @@
+require 'base64'
+require 'json'
+require 'restclient'
+require 'date'
+
+BASE_API_URI       = 'https://www.matroid.com/api/0.1'
+DEFAULT_GRANT_TYPE = 'client_credentials'
+TOKEN_RESOURCE     = '/oauth/token'
+VERBS              = %w(get post)
+
+# error messages
+NO_ENV_VARIABLES_ERR     = 'AuthenticationError: "MATROID_CLIENT_ID" and "MATROID_CLIENT_SECRET" not found in environment'
+BAD_ENV_VARIABLES_ERR    = 'AuthenticationError: problem using environment variables "MATROID_CLIENT_ID" and "MATROID_CLIENT_SECRET"'
+BAD_CLIENT_VARIABLES_ERR = 'AuthenticationError: problem using client variables provided'
```

Comment:
If you're going to have custom errors, you should subclass them from `StandardError`. The client can react to certain errors differently depending on the type without having to parse the error message.